### DISCUSSION TITLE
Is aec really needed on both pipelines?

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#3209] [Q&A] [answered]
Currently if webrtc_aec module is enabled, webrtc_aec is applied to both tx and rx pipeline:
```
audio tx pipeline:        alsa ---> aubuf ---> webrtc_aec ---> PCMU
audio rx pipeline:        alsa <--- aubuf <--- webrtc_aec <--- PCMU
```
Is it really necessary or does it help that webrtc_aec is applied also to tx pipeline?

**Comment by sreimers** [comment, marked as answer]
AEC always needs both for correct filtering/detection.

https://ez.analog.com/audio/speech-processing-solutions/w/documents/15550/acoustic-echo-cancellation
